api/src/main/java/jakarta/persistence/EntityHandler.java

```diff
@@ -678,17 +678,32 @@ <T> List<T> findMultiple(EntityGraph<T> graph, List<?> ids,
 
     /**
      * Create an instance of {@link TypedQuery} for executing a
-     * Jakarta Persistence query language statement.
-     * The select list of the query must contain only a single
-     * item, which must be assignable to the type specified by
-     * the {@code resultClass} argument.
+     * Jakarta Persistence query language SELECT statement and
+     * returning instances of the given result class. Either:
+     * <ol>
+     * <li>the select list of the query contains only a single
+     *     item, which must be assignable to the result class,
+     *     or
+     * <li>the result class must be a non-abstract class or
```

Comment:
This one goes beyond, once, in Jakarta Data, we are working on the record, if I am not mistaken...

PS: I am not against it on Jakarta Persistence.